marketplace/netlify-build/src/index.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { init } from 'contentful-ui-extensions-sdk';
+import { Select, Option, Button, Icon } from '@contentful/forma-36-react-components';
+import NeflifySideBarBuildButton from './NetlifySideBarBuildButton';
+
+import styles from './styles';
+
+const readCsvParam = csv =>
+  (csv || '')
+    .split(',')
+    .map(val => val.trim())
+    .filter(val => val.length > 0);
+
+function createSiteStructure(parameters = {}) {
+  const buildHooks = readCsvParam(parameters.buildHookUrls);
+  const siteIds = readCsvParam(parameters.netlifySiteIds);
+  const names = readCsvParam(parameters.names);
+  const urls = readCsvParam(parameters.netlifySiteUrls);
+
+  return buildHooks.reduce(
+    (acc, _, i) =>
+      acc.concat([
+        {
+          buildHookUrl: buildHooks[i],
+          channel: buildHooks[i].split('/').pop(),
+          netlifySiteId: siteIds[i],
+          name: names[i],
+          url: urls[i]
+        }
+      ]),
+    []
+  );
+}
+
+export default class NetlifyExtension extends React.Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  };
+
+  state = {
+    users: [],
+    sites: createSiteStructure(this.props.sdk.parameters.installation),
+    selectedSiteIndex: 0
+  };
+
+  async componentDidMount() {
+    this.props.sdk.window.startAutoResizer();
+    const { items } = await this.props.sdk.space.getUsers();
```

Comment:
Ah I see why they are not defined right away. I'd do a loading state, with skeleton preferably.

Reply:
I investigated this originally and found that really the only reason to have `users` is to show a user name for the messaging that happens when a build is triggered (the text that shows up under the build button). In this case, you can actually still use the application without needing to have `users`. What ends up happening is usually this loads so fast that you don't notice anything. But in the off chance that this doesn't load right away, you still see 99% of the application including the build message. It just doesn't include a username. I don't think it's worth it to make the whole thing have a loading screen to wait for this small detail.

Reply:
This being said, I'll add a test to ensure we keep this functionality the same.

Reply:
Ok, makes sense. Just make sure it looks ok if we don't have users yet

Reply:
I think it looks good. The differences are like this:

no users:
>  Triggered about 4 hours ago at 5:27:05 PM.

users:
> Triggered by David Fateh about 4 hours ago at 5:27:05 PM.

That's the only UI difference.